### PR TITLE
Problem 9

### PR DESCRIPTION
**Pull request by Achilleshiel**
Blocked by #7

### CODE REVIEW

File: filter/condition/equal.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package condition
+
+// Equal compares the received value.
+// Returns True if received is greater than or equal to threshold.
+type Equal struct {
```
Comment (silkeh) [inline filter/condition/equal.go]: Private :)

File: filter/coprime.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package filter
+
+// Coprime filters Coprime numbers
+type Coprime struct {
```
Comment (silkeh) [inline filter/coprime.go]:
```suggestion
type Coprime struct {}
```

File: filter/coprime.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	odd := true
+	for v2 := range in {
+		if odd {
+			v1 = v2
+			odd = false
+			continue
+		}
+		odd = true
+		if isCoprime(v1, v2) {
+			//fmt.Printf("Coprime: %d, %d\n",v1,v2)
```
Comment (silkeh) [inline filter/coprime.go]: No ~~capes~~ commented out code!

File: filter/pythagorean_tripleter.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package filter
+
+// PythagoreanTripleter filters palindrome numbers.
+type PythagoreanTripleter struct {
```
Comment (silkeh) [inline filter/pythagorean_tripleter.go]: Make private

File: filter/pythagorean_tripleter.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+			n, m = m, n
+		}
+		a, b, c := pt.generateTriplet(m, n)
+		if pt.isTriplet(a, b, c) {
+			for k := 1; k <= pt.n; k++ {
+				out <- a * k
+				out <- b * k
+				out <- c * k
+			}
+			//fmt.Printf("triplet: %d, %d, %d\n", a, b, c)
```
Comment (silkeh) [inline filter/pythagorean_tripleter.go]: :expressionless:

File: filter/sequence_sum.go [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package filter
+
+// SequenceSum is a filter with a sequential sum
+type SequenceSum struct {
```
Comment (silkeh) [inline filter/sequence_sum.go:4]: Make private!

File: generator/double_sequence.go [hunk 7]
```diff
@@ -0,0 +1,37 @@
+package generator
+
+// DoubleSequence generates a sequence of numbers.
+type DoubleSequence struct {
```
Comment (silkeh) [inline generator/double_sequence.go]: Make private!

**Review by silkeh — CHANGES_REQUESTED** [review]

File: filter/pythagorean_tripleter.go (hunk 5 above)
Comment (Achilleshiel) [reply filter/pythagorean_tripleter.go]: :see_no_evil:

**Review by silkeh — APPROVED** [review]

**Comment by silkeh** [comment]
LGTM, thanks!

**Pull request merged**